Formula/re2.rb

```diff
@@ -1,9 +1,9 @@
 class Re2 < Formula
   desc "Alternative to backtracking PCRE-style regular expression engines"
   homepage "https://github.com/google/re2"
-  url "https://github.com/google/re2/releases/download/2023-07-01/re2-2023-07-01.tar.gz"
-  version "20230701"
-  sha256 "18cf85922e27fad3ed9c96a27733037da445f35eb1a2744c306a37c6d11e95c4"
+  url "https://github.com/google/re2/archive/refs/tags/2023-08-01.tar.gz"
```

Comment:
Is there a reason this changed URL from attached tarball asset to generated link?

Mainly to avoid occasional risk of GitHub generated tarballs having different checksums. Right now, both tarballs are the same.

Reply:
before the last release, I think we always use the github source tarball, I can go either way.

Reply:
actually, let's maybe use the release artifact since it is available.